phpBB/memberlist.php

```diff
@@ -1469,19 +1469,58 @@
 			// Do the SQL thang
 			if ($mode == 'group')
 			{
-				$sql = "SELECT u.*
-						$sql_select
-					FROM " . USERS_TABLE . " u
-						$sql_from
-					WHERE " . $db->sql_in_set('u.user_id', $user_list) . "
-						$sql_where_data";
+				$sql_from_ary = explode(',', $sql_from);
+				$extra_tables = [];
+				foreach ($sql_from_ary as $entry)
+				{
+					$table_data = explode(' ', trim($entry));
+
+					if (empty($table_data[0]) || empty($table_data[1]))
+					{
+						continue;
+					}
+
+					$extra_tables[$table_data[0]] = $table_data[1];
+				}
+
+				$sql_array = array(
+					'SELECT'	=> 'u.*' . $sql_select,
+					'FROM'		=> array_merge([USERS_TABLE => 'u'], $extra_tables),
+					'WHERE'		=> $db->sql_in_set('u.user_id', $user_list) . $sql_where_data . '',
+				);
 			}
 			else
 			{
-				$sql = 'SELECT *
-					FROM ' . USERS_TABLE . '
-					WHERE ' . $db->sql_in_set('user_id', $user_list);
+				$sql_array = array(
+					'SELECT'	=> 'u.*',
+					'FROM'		=> array(
+						USERS_TABLE		=> 'u'
+					),
+					'WHERE'		=> $db->sql_in_set('u.user_id', $user_list),
+				);
 			}
+
+			/**
+			 * Modify user data SQL before member row is created
+			 *
+			 * @event core.memberlist_modify_memberrow_sql
+			 * @var string	mode				group
```

Comment:
It's not really the group, is it?

Reply:
I pushed the change, dunno why is not listed here below?